app/plugin.go

```diff
@@ -76,56 +77,69 @@ func (a *App) SyncPluginsActiveState() {
 			return
 		}
 
+		var wg sync.WaitGroup
 		// Deactivate any plugins that have been disabled.
 		for _, plugin := range availablePlugins {
-			// Determine if plugin is enabled
-			pluginId := plugin.Manifest.Id
-			pluginEnabled := false
-			if state, ok := config.PluginStates[pluginId]; ok {
-				pluginEnabled = state.Enable
-			}
+			wg.Add(1)
```

Comment:
I worry about expanding the parallelization beyond activation on startup to include both deactivation and general activation. I /think/ the slowest part is copying the plugins to the `plugin/` directory, and not actually activating/deactivating. 

Further inline comments, but I'd like to propose we /not/ parallelize this part unless I'm misunderstanding the effect.